corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views.js

```diff
@@ -539,13 +566,50 @@ hqDefine("cloudcare/js/formplayer/menus/views", function () {
 
                 const addressIndex = _.findIndex(this.styles, function (style) { return style.displayFormat === constants.FORMAT_ADDRESS; });
                 L.mapbox.accessToken = token;
-                const mapbox = L.mapbox;
-                const geocoder = mapbox.geocoder('mapbox.places');
 
-                const latLons = this.options.collection.models
-                    .map(model => this.addAddressPin(geocoder, addressMap, this.options.headers, model, addressIndex))
-                    .filter(latLon => latLon);
-                addressMap.fitBounds(latLons, {maxZoom: 8});
+                const latLons = []
+                const markers = []
+                this.options.collection.models
+                    .forEach(model => {
+                        const coordinates = model.attributes.data[addressIndex];
+                        if (coordinates) {
+                            let latLng = coordinates.split(" ").slice(0,2);
+                            if (latLng.length > 1) {
+                                const rowId = `row-${model.id}`;
+                                const marker = L.marker(latLng, {icon: locationIcon});
+                                markers.push(marker);
+                                marker
+                                    .addTo(addressMap)
+                                    .on('click', () => {
+                                        // tiles
+                                        $(`.list-cell-wrapper-style[id!='${rowId}']`)
+                                            .removeClass("highlighted-case");
+                                        // rows
+                                        $(`.case-row[id!='${rowId}']`)
+                                            .removeClass("highlighted-case");
+                                        $(`#${rowId}`)
+                                            .addClass("highlighted-case");
+                                        markers.forEach(m => m.setIcon(locationIcon));
+                                        marker.setIcon(selectedLocationIcon);
+                                    });
+                                latLons.push(latLng);
+                            }
+                        }
+                    });
+
+                navigator.geolocation.getCurrentPosition(
+                    success => {
+                        const coords = success.coords;
+                        const homeLatLng = [coords.latitude, coords.longitude];
+
+                        L.marker(homeLatLng, { icon: homeLocationIcon })
+                            .addTo(addressMap);
+                        latLons.push(homeLatLng);
+                        addressMap.fitBounds(latLons, {maxZoom: 8});
+                    }, () => {
+                        addressMap.fitBounds(latLons, {maxZoom: 8});
+                    }
+                );
```

Comment:
@orangejenny I tried the following but `sessionStorage.locationLat` was undefined. Debugging through formplayer showed that it did have my location though.
```
const homeLatLng = [sessionStorage.locationLat, sessionStorage.locationLong];
L.marker(homeLatLng, { icon: homeLocationIcon })
  .addTo(addressMap);
latLons.push(homeLatLng);
addressMap.fitBounds(latLons, {maxZoom: 8});
```

Reply:
> Debugging through formplayer showed that it did have my location though.

What do you mean? I do think it'd be good to understand why `sessionStorage.locationLat` wasn't working.

Reply:
ohhh, interesting - @shubham1g5 could you add a ticket to the backlog for that?

Reply:
I am not fully sure if adding a setting is a right solution here. I am wondering what does Web Apps already use `sessionStorage.locationLat` for  and if it should always populate it when required by UI instead of enabling it based on a FP flag ?

Reply:
> It looks like web apps doesn't use the location directly (excluding this PR). If formplayer requests a location ([here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/controller.js#L81-L83)), web apps will get and stick it in sessionStorage. And then every web apps request to formplayer will include the location ([here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js#L145)).

Yeah but in case of this PR, I think Web Apps should be able to override FP settings and get location if it needs it without relying on Formplayer. 

> Maybe I've just read this PR too many times and gotten used to it, but right now adding here() to the case tile template doesn't seem terrible.

Agree but I also think it's un-necessary if Web Apps can just override FP settings, although I am not sure how easy or complicated that is. 

>  Although maybe it should be added to the suite whenever a user selects the "Address" format?

The `Address` format does not necessarily need `here()` as it takes the coordinates from a case property.